src/rpy_diffusive_flux.cc

```diff
@@ -146,14 +169,18 @@ int RPYDiffusiveFlux::determineBufferShape(std::shared_ptr<State> state, const s
     {
     double max_diameter = 0;
     auto mesh = state->getMesh()->full().get();
+    
+    const GridInterpolator& g = *mobility_;
+    const double cutoff = std::get<1>(g.getBounds());
+    
     for(const auto &i : state->getTypes()) 
-    {
+        {
         const auto d_i = diameters_(i);
         if(d_i > max_diameter)
             {
             max_diameter = d_i;
             }
-    }
-    return mesh->asShape(0.5*(diameters_(type)+max_diameter));
+        }
+    return mesh->asShape(cutoff*0.5*(diameters_(type)+max_diameter));
```

Comment:
With the way the code is written, this should really just be:
```suggestion
    return mesh->asShape(cutoff);
```
and you don't need any of the for loop stuff.